File: 704. Binary Search.py
def search(nums, target):
    """
    :type nums: List[int]
    :type target: int
    :rtype: int
    """
    l, r = 0, len(nums) - 1
    while l <= r:
        t = (r - l) // 2 + l
        if target == nums[t]:
            return t
        elif target > nums[t]:
            l = t + 1
        else:
            r = t - 1
    return -1



nums = [-1,0,3,5,9,12]
target = 5

print (search(nums, target))

File: 704. Binary Search.go
package main

import "fmt"

func main() {
	nums := []int{-1, 0, 3, 5, 9, 12}
	target := 9
	fmt.Println(search(nums, target))
}

func search(nums []int, target int) int {
	l := 0
	r := len(nums) - 1
	for l <= r {
		t := (r-l)/2 + l
		if target == nums[t] {
			return t
		}
		if target > nums[t] {
			l = t + 1
		} else {
			r = t - 1
		}
	}
	return -1
}
